wrangler-core/src/main/java/io/cdap/wrangler/utils/Json2Schema.java

```diff
@@ -73,62 +73,75 @@ public Schema toSchema(String id, Row row) throws RecordConvertorException {
+    // First, we check if object is of simple type.
+    if (value instanceof String || value instanceof Integer || value instanceof Long || value instanceof Short ||
+      value instanceof Double || value instanceof Float || value instanceof Boolean || value instanceof byte[]) {
+      try {
+        return Schema.nullableOf(new SimpleSchemaGenerator().generate(value.getClass()));
+
+      } catch (UnsupportedTypeException e) {
+        throw new RecordConvertorException(
+          String.format("Unable to convert field '%s' to basic type.", name)
+        );
       }
+    }
 
-      if (value instanceof LocalDate) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof BigDecimal) {
+      // TODO CDAP-15361 precision should be derived from the schema of the row.
+      return Schema.nullableOf(Schema.decimalOf(38, ((BigDecimal) value).scale()));
+    }
 
-      if (value instanceof LocalTime) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof LocalDate) {
+      return Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
+    }
 
-      if (value instanceof ZonedDateTime) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof LocalTime) {
+      return Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS));
+    }
 
-      // TODO - remove all the instaces of java.util.Date once all the directives support LogicalType.
-      if (value instanceof Date || value instanceof java.sql.Date || value instanceof Time
-        || value instanceof Timestamp) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.Type.LONG));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof ZonedDateTime) {
+      return  Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
+    }
 
-      if (value instanceof Map) {
-        Schema schema = Schema.nullableOf (Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING)));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    // TODO - remove all the instaces of java.util.Date once all the directives support LogicalType.
+    if (value instanceof Date || value instanceof java.sql.Date || value instanceof Time
+      || value instanceof Timestamp) {
+      return Schema.nullableOf(Schema.of(Schema.Type.LONG));
+    }
 
-      if (value instanceof JsonElement) {
-        Schema schema = toSchema(name, (JsonElement) value);
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof Map) {
+      return Schema.nullableOf (Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING)));
```

Comment:
extra space after nullableOf

Reply:
Yet another reason wrangler shouldn't be examining values to determine schema. This is incorrect for maps that aren't string -> string. And there is simply no way to determine the schema if the map is empty.

You don't need to fix in this PR.